Source/SIMPLib/CoreFilters/GenerateTiltSeries.cpp

```diff
@@ -0,0 +1,665 @@
+/*
+ * Your License or Copyright can go here
+ */
+
+#include "GenerateTiltSeries.h"
+
+#include <thread>
+
+#define GTS_GENERATE_DEBUG_ARRAYS 0
+// If we are writing out all the arrays for debugging then we MUST be single threaded.
+#if(GTS_GENERATE_DEBUG_ARRAYS == 1)
+#undef SIMPL_USE_PARALLEL_ALGORITHMS
+#endif
+
+#ifdef SIMPL_USE_PARALLEL_ALGORITHMS
+#include <tbb/task_group.h>
+#include <tbb/task_scheduler_init.h>
+#endif
+
+#include "SIMPLib/Common/Constants.h"
+#include "SIMPLib/Common/SIMPLArray.hpp"
+#include "SIMPLib/FilterParameters/ChoiceFilterParameter.h"
+#include "SIMPLib/FilterParameters/DataArrayCreationFilterParameter.h"
+#include "SIMPLib/FilterParameters/DataArraySelectionFilterParameter.h"
+#include "SIMPLib/FilterParameters/FloatFilterParameter.h"
+#include "SIMPLib/FilterParameters/FloatVec3FilterParameter.h"
+#include "SIMPLib/Geometry/ImageGeom.h"
+#include "SIMPLib/Geometry/VertexGeom.h"
+#include "SIMPLib/ITK/Dream3DTemplateAliasMacro.h"
+#include "SIMPLib/ITK/SimpleITKEnums.h"
+#include "SIMPLib/ITK/itkInPlaceDream3DDataToImageFilter.h"
+#include "SIMPLib/Math/SIMPLibMath.h"
+
+#include "SIMPLib/SIMPLibVersion.h"
+
+#include <cmath>
+
+#ifndef DREAM3D_PASSIVE_ROTATION
+#define DREAM3D_PASSIVE_ROTATION 1
+#endif
+
+#ifndef ROTATIONS_CONSTANTS
+#define ROTATIONS_CONSTANTS
+namespace Rotations
+{
+namespace Constants
+{
+#if DREAM3D_PASSIVE_ROTATION
+static const float epsijk = 1.0f;
+// static const double epsijkd = 1.0;
+#elif DREAM3D_ACTIVE_ROTATION
+static const float epsijk = -1.0f;
+static const double epsijkd = -1.0;
```

Comment:
Avoid static global variables (not contained in class or function).  As of C++17, global constants should be purely const as it implies a single variable for compiler optimization, but what I've seen most commonly agreed upon is that static const global variables are inherently unsafe. The most common understanding I've seen is that each translation unit (.cpp) will have their own version of the variable.  Arguably, that's the same as const pre-17, but it looks like it's not, which can be dangerous.

I am not sure if static const globals post-17 break the assumptions for const globals, but there's a number of articles warning to not compare static const global variables with == operator due to differing memory addresses.  The pre-17 way to ensure you only have one version of a global is to use everyone's all-time favorite keyword, extern.

This is also a .cpp file, so none of that applies.  But we still should not be using static here.

Reply:
Fixed